src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -32,59 +33,70 @@ export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
 
   const stepMap: Record<LeakedPasswordsTypes, StepLink["id"]> = {
-    password: "LeakedPasswordsPassword",
-    "security-question": "LeakedPasswordsSecurityQuestion",
+    passwords: "LeakedPasswordsPassword",
+    "passwords-done": "LeakedPasswordsPassword",
+    "security-questions": "LeakedPasswordsSecurityQuestion",
+    "security-questions-done": "LeakedPasswordsSecurityQuestion",
+    none: "LeakedPasswordsSecurityQuestion",
   };
 
   const guidedExperienceBreaches = getGuidedExperienceBreaches(
     props.data.subscriberBreaches,
     props.subscriberEmails,
   );
 
+  const nextStep = getNextGuidedStep(props.data, stepMap[props.type]);
   const pageData = getLeakedPasswords({
     dataType: props.type,
     breaches: guidedExperienceBreaches,
     l10n: l10n,
+    nextStep,
   });
 
   // The non-null assertion here should be safe since we already did this check
   // in `./[type]/page.tsx`:
   const { title, illustration, content } = pageData!;
+  const isStepDone =
+    props.type === "passwords-done" || props.type === "security-questions-done";
 
   return (
     <FixView
       subscriberEmails={props.subscriberEmails}
       data={props.data}
-      nextStepHref={getNextGuidedStep(props.data, stepMap[props.type]).href}
+      nextStep={nextStep}
       currentSection="leaked-passwords"
+      hideProgressIndicator={isStepDone}
+      showConfetti={isStepDone}
     >
       <ResolutionContainer
         type="leakedPasswords"
         title={title}
         illustration={illustration}
+        isPremiumUser={hasPremium(props.data.user)}
         cta={
-          <>
-            <Button
-              variant="primary"
-              small
-              // TODO: Add test once MNTOR-1700 logic is added
-              /* c8 ignore next 3 */
-              onPress={() => {
-                // TODO: MNTOR-1700 Add routing logic
-              }}
-              autoFocus={true}
-            >
-              {l10n.getString("leaked-passwords-mark-as-fixed")}
-            </Button>
-            <Link
-              // TODO: Add test once MNTOR-1700 logic is added
-              href="/"
-            >
-              {l10n.getString("leaked-passwords-skip")}
-            </Link>
-          </>
+          !isStepDone && (
+            <>
+              <Button
+                variant="primary"
+                small
+                // TODO: Add test once MNTOR-1700 logic is added
+                /* c8 ignore next 3 */
+                onPress={() => {
+                  // TODO: MNTOR-1700 Add routing logic
+                }}
+                autoFocus={true}
+              >
+                {l10n.getString("leaked-passwords-mark-as-fixed")}
+              </Button>
+              <Link href={nextStep.href}>
+                {l10n.getString("leaked-passwords-skip")}
+              </Link>
+            </>
+          )
         }
-        estimatedTime={4}
+        estimatedTime={!isStepDone ? 4 : undefined}
```

Comment:
Since we're passing in `isStepDone`, might it make more sense to have `<ResolutionContainer>` determine whether or not to show this, based on `prop.isStepDone`?

Reply:
We are also passing `estimatedTime` with an additional condition from https://github.com/mozilla/blurts-server/blob/resolution-flow-celebration/src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/high-risk-data-breaches/HighRiskBreachLayout.tsx#L110.

Reply:
My plan has been foiled!

![image](https://y.yarn.co/7add4ae5-9709-4f23-b12d-0093c3028d53_text.gif)